### PR TITLE
Fix problems for release

### PR DESCRIPTION
**Pull request by abergeron**
These include:
- cudnn linking on mac
- sparse test failing on linux.

### CODE REVIEW

File: theano/sandbox/cuda/__init__.py [hunk 1]
```diff
@@ -362,6 +362,9 @@ def c_libraries(self):
     def c_lib_dirs(self):
         return [config.dnn.library_path]
 
+    def c_compile_args(self):
+        return ['-rpath', config.dnn.library_path]
```
Comment (nouiz) [inline theano/sandbox/cuda/__init__.py]: It is not sure the order is kept. Why not use ['-rpath '+config...] instead? Just for safety that we don't change the order of arguments

File: theano/sandbox/gpuarray/dnn.py [hunk 2]
```diff
@@ -171,6 +171,9 @@ def c_libraries(self):
     def c_lib_dirs(self):
         return [config.dnn.library_path]
 
+    def c_compile_args(self):
+        return ['-rpath', config.dnn.library_path]
```
Comment (nouiz) [inline theano/sandbox/gpuarray/dnn.py]: idem

File: theano/sandbox/gpuarray/dnn.py [hunk 3]
```diff
@@ -190,6 +193,9 @@ def c_libraries(self):
...
```
Comment (nouiz) [inline theano/sandbox/gpuarray/dnn.py]: idem

File: theano/sandbox/cuda/__init__.py (hunk 1 above)
Comment (abergeron) [inline theano/sandbox/cuda/__init__.py]: Because -rpath is a long option and can't be glued to its argument.

Comment (abergeron) [inline theano/sandbox/cuda/__init__.py]: Also, I just saw that you propose adding a space which is also a really bad idea which complicates lots of things (such as -Xcompiler processing in nvcc_compiler that will break if we do this).

Comment (nouiz) [inline theano/sandbox/cuda/__init__.py]: ok

**Comment by f0k** [comment]
c4d29a1 breaks compilation on Ubuntu 14.04.

```
gcc: error: unrecognized command line option ‘-rpath’

['nvcc', '-shared', '-O3', '-m64', '-Xcompiler', '-fno-math-errno,-Wno-unused-label,-Wno-unused-variable,-Wno-write-strings,-rpath,/usr/local/cuda/lib64,-DCUDA_NDARRAY_CUH=c72d035fdf91890f3b36710688069b2e,-DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION,-fPIC,-fvisibility=hidden', '-Xlinker', '-rpath,/run/shm/jan/.theano/compiledir_Linux-3.13--generic-x86_64-with-Ubuntu-14.04-trusty-x86_64-2.7.6-64/cuda_ndarray', '-I/usr/local/cuda/include', '-I/usr/local/lib/python2.7/dist-packages/numpy/core/include', '-I/usr/include/python2.7', '-I/raid/user/jan/install/Theano-git/theano/gof', '-I/raid/user/jan/install/Theano-git/theano/sandbox/cuda', '-o', '/run/shm/jan/.theano/compiledir_Linux-3.13--generic-x86_64-with-Ubuntu-14.04-trusty-x86_64-2.7.6-64/tmpPtQF69/265abc51f7c376c224983485238ff1a5.so', 'mod.cu', '-L/usr/lib', '-lcudnn', '-lpython2.7', '-lcudart']
```

Maybe it should only be passed to the linker. I'll investigate.

**Comment by nouiz** [comment]
the new rpath end up after -Xcompiler and not after -Xlinker. @abergeron,
can you check?

**Comment by f0k** [comment]
It actually ends up *both* after -Xlinker and after -Xcompiler. I guess it should only be used after -Xlinker. I removed it from `DnnBase` and added it to `nvcc_compiler.py:rpath_defaults` instead, but it still ends up in both places.

**Pull request merged**